tubular/wordpress_api.py

```diff
@@ -0,0 +1,27 @@
+import requests
+from requests.auth import HTTPBasicAuth
+
+
+class WordPressAPI():
+    """
+    WordPress API to retire user data.
+    """
+
+    def __init__(self, wordpress_base_url, wordpress_username, wordpress_app_password):
+        self.base_url = wordpress_base_url
+        self.username = wordpress_username
+        self.password = wordpress_app_password
+
+    def retire_user(self, learner):
+        """
+        Retires the user with the given username in the WordPress.
+        """
+
+        url = "{}/wp-json/edly/v1/retirement".format(self.base_url)
+        data = {'username': learner['original_username']}
+        response = requests.post(url, data=data, auth=HTTPBasicAuth(self.username, self.password))
+
+        if response.status_code == 200:
+            return response
+        else:
+            raise Exception(response.json()['message'])
```

Comment:
what if our response is None? it will not provide the right exception message instead key not found issue will be raised

Reply:
The API always return some response even there is some error. In case some error occur the whole function is wrap in `try` block https://github.com/edly-io/tubular/blob/master/tubular/scripts/retire_one_learner.py#L183